multipool-tracker/src/multipool_storage/mod.rs

```diff
@@ -53,17 +56,19 @@ const X96: u128 = 96;
 const X32: u128 = 32;
 
 impl MultipoolAsset {
-    fn quoted_quantity(&self) -> Option<MayBeExpired<Price>> {
-        merge(
-            self.quantity_slot.clone(),
-            self.price.clone(),
-            |slot, price| {
-                slot.merge(price, |slot, price| -> Option<U256> {
-                    slot.quantity.checked_mul(price).map(|m| m.shr(X96))
-                })
-                .transpose()
-            },
-        )
+    fn quoted_quantity(&self) -> Result<MayBeExpired<Price>, MultipoolErrors> {
+        if let (Some(slot), Some(price)) = (self.quantity_slot.clone(), self.price.clone()) {
```

Comment:
We shouldn't send FieldMissing as it is better to specify different QuantitySlotMissing or PriceMissing with specified asset. This approach is better because it allows to determine which data is unset and to potentially handle this differently, moreover this kind of error should contain asset address. In context of this function it makes no sense as you specify the only fetched asset address in function args. However it will be useful when this error is propagated later with ? in other storage functions